File: PWRSyllabus.API/ClientApp/src/models/SubjectCard.ts
import Course from './Course';
import FieldOfStudy from './FieldOfStudy';
import {Employee, DefaultEmployee} from './Employee';
import EducationalEffect from '@/models/EducationalEffect';


export default interface SubjectCard {
    id: number;
    subjectCode: string;
    nameInPolish: string;
    nameInEnglish: string;
    specialization: string;
    level: string;
    formOfStudies: string;
    isGroupOfCourses: boolean;
    subjectType: string;
    prerequisites: string[];
    objectivities: string[];
    courses: Course[];
    supervisor: Employee; // !!! ???
    educationalEffects: EducationalEffect[];
    teachingTools: string[];
    primaryLiterature: string[];
    secondaryLiterature: string[];
}

const DefaultSubjectCard: SubjectCard = {
    id: 0,
    subjectCode: '',
    nameInPolish: '',
    nameInEnglish: '',
    specialization: '',
    level: '6',
    formOfStudies: '',
    isGroupOfCourses: false,
    subjectType: '',
    prerequisites: [],
    objectivities: [],
    courses: [],
    supervisor: DefaultEmployee, // !!! ???
    educationalEffects: [],
    teachingTools: [],
    primaryLiterature: [],
    secondaryLiterature: []
};

export {DefaultSubjectCard, SubjectCard};

File: PWRSyllabus.Core/Entities/SubjectCard.cs
using System.Collections.Generic;
using PWRSyllabus.Core.Enums;

namespace PWRSyllabus.Core.Entities
{
    public class SubjectCard : BaseEntity
    {
        public string NameInPolish { get; set; }
        public string NameInEnglish { get; set; }
        public string SubjectCode { get; set; }
        public bool IsGroupOfCourses { get; set; }
        public string Prerequisites { get; set; }
        public string Objectivities { get; set; }
        public string PrimaryLiterature { get; set; }
        public string SecondaryLiterature { get; set; }
        public string TeachingTools { get; set; }
        public int Version { get; set; }
        public SubjectType SubjectType { get; set; }
        public CourseForm? FinalCourseForm { get; set; }
        public Employee Supervisor { get; set; }
        public int SupervisorId { get; set; }
        public List<StudyPlanSubjectCard> StudyPlanSubjectCards { get; set; }
        public List<StudyProgramSubjectCard> StudyProgramSubjectCards { get; set; }
        public List<EducationalEffectSubjectCard> EducationalEffectSubjectCards { get; set; }
        public List<Course> Courses { get; set; }
        public FormOfStudies FormOfStudies { get; set; }
    }
}
